src/schema/organization/contributors/resolvers.js

```diff
@@ -1,61 +1,131 @@
+ */
+export class ContributorSet {
+  constructor() {
+      this.map = new Map();
+      this[Symbol.iterator] = this.values;
+  }
+
+  add(item) {
+    this.map.set(item.login, item);
+  }
+
+  values() {
+      return this.map.values();
+  }
+
+  delete(item) {
+      return this.map.delete(item.login);
+  }
+}
 
 /**
  * Process the repositories array to leave only the information the schema needs.
  * @param {Array} repositories 
  * @return processed repositories array
  */
-const process = (repositories) => {
-  const contributors = new GeneralSet();
+const processContributors = (repositories) => {
+  const contributors = new ContributorSet();
 
   repositories.forEach(repository => {
     repository.mentionableUsers.nodes.forEach(user => {
-      const firstPullRequest = user.contributionsCollection.pullRequestContributions.nodes[0];
-      const firstIssue = user.contributionsCollection.issueContributions.nodes[0];
-      const firstReview = user.contributionsCollection.pullRequestReviewContributions.nodes[0];
-
-      let currentDate = undefined;
-
-      if (firstPullRequest || firstIssue || firstReview) {
-        const firstContributions = [firstPullRequest, firstIssue, firstReview];
-        const dates = [];
-        
-        firstContributions.forEach(contribution => {
-          if (contribution) dates.push(contribution.firstContributionDate);
-        });
-
-        currentDate = getOldestDate(dates);
-      }
-
-      user.contributionsCollection = undefined;
-      user.firstContributionDate = currentDate;
+      user.a = 1;
       contributors.add(user);
     });
   });
   
   return Array.from(contributors.values());
 };
 
+/**
+ * Process the user object and returns the Date of the first contribution or undefined
+ * if the user has no contributions.
+ * @param {Object} user 
+ * @return {Date} first contribution date
+ */
+const processFirstContributionDate = (user) => {
+  if (user) {
+    const issuesNodes = user.contributionsCollection.issueContributions.nodes;
+    const pullRequestsNodes = user.contributionsCollection.pullRequestContributions.nodes;
+    const reviewsNodes = user.contributionsCollection.pullRequestReviewContributions.nodes;
+    const contributions = [issuesNodes, pullRequestsNodes, reviewsNodes];
+
+    const dates = [];
+    var firstContrib;
+    contributions.forEach(contribution => {
+      if (contribution && contribution.length > 0) {
+        firstContrib = contribution[0].occurredAt;
+        dates.push(new Date(firstContrib)); 
+      }
+    });
+
+    var result = undefined;
+    if (dates.length > 0)
+      result = getOldestDate(dates);
+    return result;
+  }
+}
+
+const firstContributionDateResolver = async(parent, args, { token }) => {
+  const { login } = args;
+  const body = githubQueries.firstContributionDate(login, parent.id);
+  const data = await fetchData(body, token);
+  
+  var result = processFirstContributionDate(data.data.user);
```

Comment:
Don't use `var`, use `const` please

Reply:
It was code remnants from other implementation, sorry